Assets/Scripts/Buildings/Building.cs

```diff
@@ -7,15 +7,15 @@ namespace Buildings
     public class Building : Damageable
     {
         [SerializeField] private GameObject destructionParticleEffect;
-        [SerializeField] private int maxLevel;
 
-        public int Level { get; private set; }
-
-        private void Start()
-        {
-            Level = 1;
-            CurrentHealth = maxHealth;
-        }
+        public int MaxLevel { get; set; }
+        
+        private int _currentLevel;
+        
+        public delegate void Upgrade(int level);
+        public event Upgrade OnUpgrade;
+        
+        private void Start() => CurrentHealth = maxHealth;
```

Comment:
Passiert das nich schon in Damageable?

Reply:
![](https://thumbs.gfycat.com/WebbedFairAmazondolphin-max-1mb.gif)